internal/observer/prometheus/metrics.go

```diff
@@ -137,3 +137,140 @@ func BuildMemoryLimitsQuery(labelFilter string) string {
             kube_pod_labels{%s}
         )`, labelFilter)
 }
+
+// ----------------------------
+// HTTP REQUEST METRICS QUERIES
+// ----------------------------
+
+// BuildHTTPRequestCountQuery builds a PromQL query for HTTP request count
+func BuildHTTPRequestCountQuery(labelFilter string) string {
+	return fmt.Sprintf(`
+	    rate(hubble_http_requests_total{reporter="client"}[2m])
+            * on(destination_pod) group_left(label_openchoreo_dev_component_uid, label_openchoreo_dev_project_uid, label_openchoreo_dev_environment_uid) 
+            label_replace(
+	            kube_pod_labels{%s},
+    			"destination_pod",
+	    		"$1",
+		    	"pod",
+			    "(.*)"
+		    ) 
+			>= 0
+	`, labelFilter)
+}
+
+// BuildSuccessfulHTTPRequestCountQuery builds a PromQL query for successful HTTP request count. Requests are
+// considered successful if they have a 200 status code.
+func BuildSuccessfulHTTPRequestCountQuery(labelFilter string) string {
+	return fmt.Sprintf(`
+	    rate(hubble_http_requests_total{reporter="client", status="200"}[2m])
```

Comment:
any 1xx-3xx request should be considered successful, you can use the regex filter for this.